### ISSUE TITLE
Allow drag and drop

### ISSUE THREAD

**Issue by koppor** [liulex/Snipaste-Feedback#416]
I'd like to drop the screenshot as filesomewhere and that snipaste deletes the file after closing. This is a functionality of grab'n'drop (see http://softwarerecs.stackexchange.com/a/25130/18184), but Snipaste is more powerful in other regards.

Here the screenshot how Grab'n'Drop does it:

![grafik](https://cloud.githubusercontent.com/assets/1366654/24944343/0412b81e-1f5b-11e7-922b-f20d4fe850d1.png)

The main window lets you specify a filename. If that window looses the focus, a small window appears. This window shows a minified version of the screenshot. One can drag'n'drop from here and it always pastes a file.

This is similar to https://github.com/liulex/Snipaste-Feedback/issues/282#issuecomment-268241123 (#78), but I still miss the drag'n'drop feature.

I really like the automatic file deletion at this feature of Grab'n'Drop: When closing the application, the file is deleted. The reason is that I use screenshots for posting into Skype, Emails, posting into GitHub and Slack.

Currently, I can do a similar functionality using the Clipboard feature. However, the filename is always `grafik.png`. 

Furthermore, I cannot paste into the file explorer. So, for each target application, I have to think, whether I should use clipboard or file export. If I want to determine the filename for an uploaded file, I have to save it somewhere, rename it and attach it again.

With the grab'n'drop functionality, I can work much faster.

In Summary:

- [ ] specify filename for file (in the clipboard)
- [ ] allow pasting into file explorer
- [ ] small window allowing drag and drop
- [ ] drag area (with filename field) in the main window

**Comment by liulex** [comment]
Thanks for your detailed desciption. And also thank you for recommending Snipaste on [StackExchange](http://softwarerecs.stackexchange.com/a/25130/18184). ❤️ 

I would like to try GrabNDrop so I downloaded it from [here](https://github.com/scyv/GrabNDrop/releases) and [here](http://www.softpedia.com/get/Multimedia/Graphic/Graphic-Capture/GrabNDrop.shtml) (the [official site](http://www.scyv.de/gnd/) seems broken). But when I pressed <kbd>Enter</kbd> after screenshot, I got this:
![default](https://cloud.githubusercontent.com/assets/2010459/24951595/84e4da7a-1fa7-11e7-9166-86e0d25bf2fc.png)

I tried two computers (both Win10 x64) with the same result. Could you [send me](mailto:snipaste.app@gmail.com) a copy that works?

**Comment by koppor** [comment]
Demo using main window of grab'n'drop:

![main-window](https://cloud.githubusercontent.com/assets/1366654/25223385/8dd9e0ac-25bb-11e7-9e0b-d4fef755c977.gif)

Demo using the additional model dialog:

![modal-window gif](https://cloud.githubusercontent.com/assets/1366654/25223579/051b5f88-25bc-11e7-96c0-a7d27978d07c.gif)

**Comment by koppor** [comment]
My main use case for that is sending a screenshot via Email in Outlook with plain-text emails.

A pasting from the clipboard to plain-text-emails leads to the error "pictures will get lost".

**Comment by koppor** [comment]
I am really a fan of this feature. Would a donation help to get this feature implemented?

**Comment by koppor** [comment]
Since <kbd>Ctrl</kbd>+<kbd>V</kbd> works, I don't need this feature any more.
